### PR TITLE
Improve exception messages referencing startup code

### PR DESCRIPTION
**Pull request by halter73**
- Remove "ConfigureServices(...)" reference in exception message
- Improve exception message for missing IAuthenticationService

Addresses #37778

I'll backport this to `release/6.0` once we agree on the exception message.

### CODE REVIEW

File: src/Http/Authentication.Abstractions/src/Resources.resx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    <data name="Icon1" type="System.Drawing.Icon, System.Drawing" mimetype="application/x-microsoft.net.object.bytearray.base64">
+        <value>[base64 mime encoded string representing a byte array form of the .NET Framework object]</value>
+        <comment>This is a comment</comment>
+    </data>
+                
+    There are any number of "resheader" rows that contain simple 
+    name/value pairs.
+    
+    Each data row contains a name, and value. The row also contains a 
+    type or mimetype. Type corresponds to a .NET class that support
```
Comment (Rick-Anderson) [inline src/Http/Authentication.Abstractions/src/Resources.resx:34]: Pick on of these, I prefer the first suggestion:

```suggestion
    Each data row contains a name, a value, and a
    type or mimetype. Type corresponds to a .NET class that support 
```


```suggestion
    Each data row contains a name and value. The row also contains a 
    type or mimetype. Type corresponds to a .NET class that support 
```

File: src/Http/Authentication.Abstractions/src/Resources.resx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    There are any number of "resheader" rows that contain simple 
+    name/value pairs.
+    
+    Each data row contains a name, and value. The row also contains a 
+    type or mimetype. Type corresponds to a .NET class that support 
+    text/value conversion through the TypeConverter architecture. 
+    Classes that don't support this are serialized and stored with the 
+    mimetype set.
+    
+    The mimetype is used for serialized objects, and tells the
```
Comment (Rick-Anderson) [inline src/Http/Authentication.Abstractions/src/Resources.resx:39]:
```suggestion
    The mimetype is used for serialized objects and tells the 
```

File: src/Http/Authentication.Abstractions/src/Resources.resx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    text/value conversion through the TypeConverter architecture. 
+    Classes that don't support this are serialized and stored with the 
+    mimetype set.
+    
+    The mimetype is used for serialized objects, and tells the 
+    ResXResourceReader how to depersist the object. This is currently not 
+    extensible. For a given mimetype the value must be set accordingly:
+    
+    Note - application/x-microsoft.net.object.binary.base64 is the format 
+    that the ResXResourceWriter will generate, however the reader can
```
Comment (Rick-Anderson) [inline src/Http/Authentication.Abstractions/src/Resources.resx:44]:
```suggestion
    that the ResXResourceWriter generates, however the reader can 
```

**Review by Rick-Anderson — APPROVED** [review]

**Review by JamesNK — APPROVED** [review]

**Review by HaoK — APPROVED** [review]

**Comment by Kahbazi** [comment]
Do these messages need to be updated to remove `Configure(..) `?
https://github.com/dotnet/aspnetcore/blob/8b30d862de6c9146f466061d51aa3f1414ee2337/src/Http/Routing/src/EndpointMiddleware.cs#L93
https://github.com/dotnet/aspnetcore/blob/8b30d862de6c9146f466061d51aa3f1414ee2337/src/Http/Routing/src/EndpointMiddleware.cs#L101

Here's the tests for the said messages.
https://github.com/dotnet/aspnetcore/blob/8b30d862de6c9146f466061d51aa3f1414ee2337/src/Http/Routing/test/UnitTests/EndpointMiddlewareTest.cs#L201
https://github.com/dotnet/aspnetcore/blob/8b30d862de6c9146f466061d51aa3f1414ee2337/src/Http/Routing/test/UnitTests/EndpointMiddlewareTest.cs#L104

File: src/Http/Authentication.Abstractions/src/Resources.resx [hunk 4]
```diff
@@ -0,0 +1,123 @@
+<?xml version="1.0" encoding="utf-8"?>
+<root>
+  <!-- 
+    Microsoft ResX Schema 
+    
+    Version 2.0
+    
+    The primary goals of this format is to allow a simple XML format
```
[the remark this answers is not in the document]
Comment (halter73) [reply src/Http/Authentication.Abstractions/src/Resources.resx:8]: This is generated by VS, so I'm leaving it as is.

**Comment by halter73** [comment]
I'm not going to re-request approvals but I changed the EndpointMiddleware messages too along the same lines. [`9aa2098` (#38045)](https://github.com/dotnet/aspnetcore/pull/38045/commits/9aa2098f32e7e205cca4e91f1acd7444a076230b#diff-057b120afd36eaa61be807328150289e2033c7193ab14800e252c32c56665c69L93)

**Review by Tratcher — APPROVED** [review]

**Pull request merged**